cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -384,8 +385,12 @@ func (g *genDeepCopy) Init(c *generator.Context, w io.Writer) error {
 	sw.Do("if err := $.scheme|raw$.AddGeneratedDeepCopyFuncs(\n", map[string]interface{}{
 		"scheme": scheme,
 	})
+	reflect := c.Universe.Type(types.Name{Package: "reflect", Name: "TypeOf"})
+	runtime := c.Universe.Type(types.Name{Package: runtimePackagePath, Name: "GeneratedDeepCopyFunc"})
 	for _, t := range g.typesForInit {
-		sw.Do(fmt.Sprintf("%s,\n", g.funcNameTmpl(t)), argsFromType(t))
+		g.imports.AddType(reflect)
+		g.imports.AddType(runtime)
+		sw.Do(fmt.Sprintf("runtime.GeneratedDeepCopyFunc{Fn: %s, InType: reflect.TypeOf(func() *$.type|raw$ {var x *$.type|raw$; return x}())},\n", g.funcNameTmpl(t)), argsFromType(t))
```

Comment:
Not sure I can follow you here. The lookup in the cloner is the same as before, with the exception that the key used is the pointer type. Deepcopies of non-pointer types will fall back to reflection based cloning. In my local checkout I have a bold message `FIXME, THIS IS SLOW using reflection for type %v`. In other words, those non-pointer deepcopies are not supposed to be done. To enforce this with static type checking we could change the signature of the public `DeepCopy` functions into:

```
func DeepCopy(in *runtime.Object) (*runtime.Object, error)
```